Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/ObjCStubs.kt

```diff
@@ -256,18 +256,30 @@ internal fun Type.isStret(target: KonanTarget): Boolean {
         KonanTarget.MACOS_X64,
         KonanTarget.WATCHOS_X64,
         KonanTarget.TVOS_X64 -> when (unwrappedType) {
-            is RecordType -> unwrappedType.decl.def!!.size > 16 || this.hasUnalignedMembers()
+                is RecordType -> unwrappedType.decl.def!!.size > 16 || this.hasUnalignedMembers()
+                else -> false
+            }
+
+        // See: https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/LowLevelABI/130-IA-32_Function_Calling_Conventions/IA32.html#//apple_ref/doc/uid/TP40002492-SW5
+        // Structures 1 or 2 bytes in size are placed in EAX.
+        // Structures 4 or 8 bytes in size are placed in: EAX and EDX.
+        // Structures of other sizes are placed at the address supplied by the caller.
+        KonanTarget.WATCHOS_X86 -> when (unwrappedType) {
+            is RecordType -> {
+                val size = unwrappedType.decl.def!!.size
+                val canBePassedInRegisters = (size == 1L || size == 2L || size == 4L || size == 8L)
+                return !canBePassedInRegisters || this.hasUnalignedMembers()
```

Comment:
This seems wrong:
```c
struct __attribute__((packed)) S {
    char x;
    short y;
    char z;
};
```

Reply:
Generally this ad hoc ABI looks like a hack, which we can tolerate now, but likely shall have better solution in the future.